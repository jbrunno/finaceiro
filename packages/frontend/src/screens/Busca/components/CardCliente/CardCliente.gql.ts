import { gql } from '@apollo/client';
import { ClienteTipoEnum } from '@frontend/core/dist/types';

export const CARD_CLIENTE_FRAGMENT = {
  cliente: gql`
    fragment CARD_CLIENTE_FRAGMENT on Cliente {
      id
      nome
      documento
      falecido
      negativado
      nomeSocial
      tipo
    }
  `,
};

export type ClienteData = {
  id: string;
  nome: string;
  documento: string;
  falecido: boolean;
  negativado: boolean;
  nomeSocial: string;
  tipo: ClienteTipoEnum;
};
