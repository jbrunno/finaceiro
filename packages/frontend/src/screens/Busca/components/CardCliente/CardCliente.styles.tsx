import { Card, CardContent, Chip } from '@frontend/components';
import { css, styled } from '@frontend/styles';

export const CardContentStyled = styled(CardContent)<{
  position: 'top' | 'bottom';
}>(
  ({ theme, position }) => css`
    padding: ${theme.fn.spacing(16)};
    border-top: ${position === 'bottom' &&
    `1px solid ${theme.palette.divider}`};
    border-bottom: ${position === 'top' &&
    `1px solid ${theme.palette.divider}`};
  `,
);

export const ChipStyled = styled(Chip)`
  margin-left: ${({ theme }) => theme.fn.size(8)};
`;

export const CardStyled = styled(Card)`
  max-height: ${({ theme }) => theme.fn.size(218)};
`;
