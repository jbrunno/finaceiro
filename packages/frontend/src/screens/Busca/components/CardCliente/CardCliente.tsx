import {
  CardText,
  Chip,
  Grid,
  Typography,
  CardHeaderChip,
  RenderComponents,
} from '@frontend/components';
import { formatDocument } from '@frontend/utils';
import {
  CardContentStyled,
  CardStyled,
  ChipStyled,
} from './CardCliente.styles';
import { CARD_CLIENTE_FRAGMENT, ClienteData } from './CardCliente.gql';

type CardClienteProp = {
  onClick?: () => void;
} & ClienteData;

export function CardCliente({
  nome,
  nomeSocial,
  documento,
  falecido,
  tipo,
  negativado,
  onClick,
}: CardClienteProp) {
  return (
    <CardStyled onClick={onClick}>
      <CardHeaderChip
        title={
          <Typography variant="h5" color="textPrimary">
            {nomeSocial || nome}
          </Typography>
        }
        hasDivider
        chip={
          <Grid>
            {negativado && <Chip label="Negativado" color="error" />}
            {falecido && <ChipStyled label="Falecido" />}
          </Grid>
        }
      />
      <CardContentStyled position="bottom">
        <RenderComponents screen="busca">
          <Grid marginBottom={2} w-perm="CLIENTE_NOME_REGISTRO">
            <CardText label="Nome de registro" text={nome} />
          </Grid>
          <Grid w-perm="CLIENTE_DOCUMENTO">
            <CardText
              label="CPF/CNPJ"
              text={documento ? formatDocument(documento) : '-'}
            />
          </Grid>
          <Grid w-perm="CLIENTE_DOCUMENTO_INTERNACIONAL">
            <CardText
              label="CPF/CNPJ/ID ou Passaporte"
              text={documento ? formatDocument(documento, tipo) : '-'}
            />
          </Grid>
        </RenderComponents>
      </CardContentStyled>
    </CardStyled>
  );
}

CardCliente.fragments = CARD_CLIENTE_FRAGMENT;
