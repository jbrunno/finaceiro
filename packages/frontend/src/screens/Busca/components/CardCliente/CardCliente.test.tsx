import React from 'react';
import { render } from '@test-utils/component-testing';
import { CardCliente } from './CardCliente';
import { ClienteData } from './CardCliente.gql';

test('renders the CardCustomer', () => {
  const cliente = {
    nome: 'João da Silva',
    nomeSocial: 'João',
    documento: '123.456.789-00',
  } as ClienteData;
  const { container } = render(<CardCliente {...cliente} />);
  expect(container).toMatchSnapshot();
});
