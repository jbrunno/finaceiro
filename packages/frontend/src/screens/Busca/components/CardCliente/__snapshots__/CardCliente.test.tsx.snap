// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the CardCustomer 1`] = `
.emotion-2 {
  background-color: #fff;
  color: #3B3B41DE;
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  overflow: hidden;
  overflow: visible;
  border-radius: 0.5rem;
  box-shadow: 0px 2px 4px -1px rgba(233, 233, 233, 0.8),0px 4px 5px rgba(233, 233, 233, 0.14),0px 1px 10px rgba(233, 233, 233, 0.12);
  background-color: #fff;
  max-height: calc(9.5375rem + 65.39999999999999px);
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 16px;
  display: grid;
  grid-template-columns: 1fr auto auto;
  grid-template-areas: 'content avatar action';
  border-bottom: 1px solid #0000001F;
}

.emotion-4 .MuiCardHeader-content {
  grid-area: content;
}

.emotion-4 .MuiCardHeader-avatar {
  margin-right: 0;
  margin-left: 0.5rem;
  grid-area: avatar;
}

.emotion-4 .MuiCardHeader-avatar>div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-4 .MuiCardHeader-action {
  grid-area: action;
  margin: auto;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  margin-right: 16px;
}

.emotion-6 {
  box-sizing: border-box;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-7 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-8 {
  margin: 0;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.125rem;
  letter-spacing: 0.0094rem;
  font-family: Lato,sans-serif;
  display: block;
}

.emotion-9 {
  display: grid;
  -webkit-column-gap: 0.25rem;
  column-gap: 0.25rem;
  white-space: nowrap;
  grid-template-columns: auto 1fr;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-column-gap: 0.25rem;
  column-gap: 0.25rem;
}

.emotion-13 {
  margin: 0;
  font-weight: 700;
  font-size: 1rem;
  line-height: 1.5rem;
  letter-spacing: 0;
  font-family: Lato,sans-serif;
  color: #3B3B41DE;
}

.emotion-15 {
  padding: 16px;
  padding: 1rem;
  border-top: 1px solid #0000001F;
}

.emotion-15:last-child {
  padding-bottom: 24px;
}

<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-0 emotion-1 emotion-2"
  >
    <div
      class="MuiCardHeader-root emotion-3 emotion-4"
    >
      <div
        class="MuiCardHeader-avatar emotion-5"
      >
        <div
          class="MuiGrid-root emotion-6"
        />
      </div>
      <div
        class="MuiCardHeader-content emotion-7"
      >
        <span
          class="MuiTypography-root MuiTypography-body2 MuiCardHeader-title emotion-8"
        >
          <div
            class="emotion-9 emotion-10"
          >
            <div
              class="emotion-11 emotion-12"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5 emotion-13"
              >
                Jo√£o
              </h5>
            </div>
          </div>
        </span>
      </div>
    </div>
    <div
      class="MuiCardContent-root emotion-14 emotion-15"
      position="bottom"
    />
  </div>
</div>
`;
